### DISCUSSION TITLE
How to have env variable interpreted on run command

### DISCUSSION THREAD

**Discussion by pixsaoul** [community/community#25026] [Actions]
<p>Hello, </p>
<p>I am new to github actions and I’m having trouble using env variables.</p>
<p>What I would like :</p>
<pre><code>jobs:
  test:
    runs-on: [self-hosted, windows, x64]

    steps:
    - name: Tests
      env:
        testVariable: tototh
      run: echo ${testVariable}
</code></pre>
<p>but I cannot get echo to actually display tototh as it does if I put directly “echo tototh”</p>
<p>am I missing something ? I tried with multiple syntax for variables like ${{ testVariable }} with the same result.</p>
<p>Am I missing something obvious ? </p>
<p>I am using a self hosted runner on windows (as the code suggests).</p>
<p>I saw this topic brought up about secret variables but not any env var like this.</p>
<p>Ultimately I’d like to use job defined env and/or workflow define env. Is there anything more to add to use it ?</p>
<p>Thank you for your time =)</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi pixsaoul, </p>
<p>As you are using self-hosted runner on windows , windows system use PowerShell as default shell . Please check your logs, does it use C:\Program Files\PowerShell\6\pwsh.EXE or C:\windows\System32\WindowsPowerShell\v1.0\powershell.EXE? </p>
<p>You need to use ${env: testvariable} to interprete env in powershell script. </p>
<p>Please see my result: </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2113815529c02219c832ed24da7b4cdb75be704a.png" data-download-href="/uploads/short-url/4IBxRJbrcz2LCnEnCo7Fg2BQatI.png?dl=1" title="Powershell script.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181088488-405d29d3-e423-4122-82c1-ccbe30935e14.png" alt="Powershell script.png" data-base62-sha1="4IBxRJbrcz2LCnEnCo7Fg2BQatI" width="690" height="357" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/2113815529c02219c832ed24da7b4cdb75be704a_2_690x357.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/2113815529c02219c832ed24da7b4cdb75be704a_2_1035x535.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/2/2113815529c02219c832ed24da7b4cdb75be704a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2113815529c02219c832ed24da7b4cdb75be704a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Powershell script.png</span><span class="informations">1124×583 30.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>There is the document for using specific shell and the corresponding env variable interprete syntax. <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#using-a-specific-shell" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#using-a-specific-shell </a></p>
